kubernetes/client_factory.go

```diff
@@ -144,22 +166,35 @@ func (cf *clientFactory) newClient(authInfo *api.AuthInfo, expirationTime time.D
 	return newClient, err
 }
 
+func (cf *clientFactory) newSAClient(cluster string) (*K8SClient, error) {
+	// TODO: Need a way to load in cluster configuration from the kube secret.
```

Comment:
I guess this `TODO` means that this `newSAClient` func isn't really capable of creating a client for remote clusters, right?.

If so, I guess that, although we can say that the cache would be ready for MC after merging this PR, the client factory is not ready yet.

Reply:
That's right. The cache will start a kube cache for each kiali service account client from the client factory but the client factory only supports a single service account client for now. It will need to be updated to read in the remote secrets from the filesystem and create a kiali service account client for each remote secret. This can happen after #5752 is merged.